legion_test/bin/cleanup_nexus_repo

```diff
@@ -0,0 +1,200 @@
+
+    :param api_url: Nexus API URL
+    :type api_url: str
+    :param user: Nexus admin user
+    :type user: str
+    :param password: Nexus admin user password
+    :type password: str
+    :param builds: List of Nexus components
+    :type builds: list
+    :param dry_run: Dry run mode selector, no delete actions to be performed if True
+    :param dry_run: bool
+    :return: None
+    """
+    log.info("List of artifacts to delete: {}".format(
+        [("{}-{}".format(build.get("name"), build.get("version"))) for build in builds]))
+    if dry_run:
+        log.info("Dry run mode selected, no actions to be performed.")
+    else:
+        for build in builds:
+            try:
+                requests.delete(NEXUS_DELETE_URL.format(
+                    host=nexus_url,
+                    resource=build.get("id")),
+                    auth=(user, password))
+                log.info("{}-{}-{}: Deleted".format(
+                    build.get("name"),
+                    build.get("version"),
+                    build.get("id")))
+            except Exception as e:
+                log.error(e)
+
+
+def work(options):
+    """
+    Get Legion artifacts matching build_name (if specified) from Nexus repository
+    and delete those who older than remaining_days
+
+    :param options: options
+    :type options: :py:class:`Options`
+    :return: None
+    """
+
+    for repo in options.repositories:
+        log.info('Cleaninig up {} Nexus repository'.format(repo))
+        try:
+            builds_to_del = filter_builds_by_date(get_builds(
+                options.nexus_url,
+                repo,
+                options.user,
+                options.password,
+                options.build_name), options.remaining_days)
+            try:
+                delete_builds(
+                    options.nexus_url,
+                    options.user,
+                    options.password,
+                    builds_to_del,
+                    options.dry_run)
+            except Exception as e:
+                log.error('Can\'t delete artifacts for {} repository: {}'.format(repo, e))
+        except Exception as e:
+            log.error('Error fetching list of buils for {} repository: {} '.format(repo, e))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Script for Legion artifacts cleanup at Nexus repository')
+    parser.add_argument('nexus_url', type=str, help='Nexus repo url')
+    parser.add_argument('--user', '-u', type=str, default='admin', help='Nexus admin user')
+    parser.add_argument('--password', '-p', type=str, default='admin123', help='Nexus admin password')
+    parser.add_argument('--remaining-days', type=int, help='Number of days to remain builds')
+    parser.add_argument('--build-name', '-n', type=str, help='Build name to delete')
+    parser.add_argument('--repositories', '-r', type=str, default=['main-docker', 'jenkins_plugins', 'pypi-hosted'],
```

Comment:
Are you sure that this works? With just type=str

Reply:
fixed